corehq/apps/sms/api.py

```diff
@@ -458,31 +492,12 @@ def process_sms_registration(msg):
                         keyword3 in REGISTRATION_MOBILE_WORKER_KEYWORDS
                         and domain_obj.sms_mobile_worker_registration_enabled
                 ):
-                    if keyword4 != '':
-                        username = keyword4
-                    else:
-                        username = cleaned_phone_number
-                    try:
-                        user_data = {}
-
-                        username = process_username(username, domain_obj)
-                        password = random_password()
-                        new_user = CommCareUser.create(domain_obj.name, username, password, created_by=None,
-                                                       created_via=USER_CHANGE_VIA_SMS, metadata=user_data)
-                        new_user.add_phone_number(cleaned_phone_number)
-                        new_user.save()
-
-                        entry = new_user.get_or_create_phone_entry(cleaned_phone_number)
-                        entry.set_two_way()
-                        entry.set_verified()
-                        entry.save()
-                        registration_processed = True
-
-                        if domain_obj.enable_registration_welcome_sms_for_mobile_worker:
-                            send_sms(domain_obj.name, None, cleaned_phone_number,
-                                     get_message(MSG_REGISTRATION_WELCOME_MOBILE_WORKER, domain=domain_obj.name))
-                    except ValidationError as e:
-                        send_sms(domain_obj.name, None, cleaned_phone_number, e.messages[0])
+                    registration_processed = register_sms_user(
+                        username=cleaned_phone_number if keyword4 == '' else keyword4,
```

Comment:
nit: I see we have flipped the logic here which should be safe but just scares me a bit with comparisons like None with an empty string. 
`username=keyword4 if keyword4 != '' else cleaned_phone_number` would be the exact same.